modular/blocksyncer/modules/payment/payment_handler.go

```diff
@@ -92,3 +104,31 @@ func (m *Module) handleEventStreamRecordUpdate(ctx context.Context, streamRecord
 		k: v,
 	}
 }
+
+func (m *Module) handleEventBucketFlowRateLimitStatus(ctx context.Context, block *tmctypes.ResultBlock, txHash common.Hash, bucketFlowRateLimitStatus *storagetypes.EventBucketFlowRateLimitStatus) map[string][]interface{} {
+	bucket, err := m.db.GetBucketByBucketName(ctx, bucketFlowRateLimitStatus.BucketName)
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return nil
+		}
+	}
+	var offChainStatus int
+	if bucketFlowRateLimitStatus.IsLimited {
+		offChainStatus = bucketmodule.AddStatus(bucket.OffChainStatus, int(bucketmodule.OffChainStatusIsLimited))
+	} else {
+		offChainStatus = bucketmodule.RemoveStatus(bucket.OffChainStatus, int(bucketmodule.OffChainStatusIsLimited))
+	}
+	bucketStatus := &models.Bucket{
+		BucketName:     bucketFlowRateLimitStatus.BucketName,
+		OffChainStatus: offChainStatus,
+
+		UpdateAt:     block.Block.Height,
+		UpdateTxHash: txHash,
+		UpdateTime:   block.Block.Time.UTC().Unix(),
+	}
+
+	k, v := m.db.UpdateBucketByNameToSQL(ctx, bucketStatus)
```

Comment:
This might have some issue, as the offChainStatus might be change back to 0 some day, and UpdateBucketByNameToSQL use gorm default Updates will not handle default values.